dask/order.py

```diff
@@ -381,6 +388,42 @@ def process_runnables() -> None:
     # writing, the most expensive part of ordering is the prep work (mostly
     # connected roots + sort_key) which can be reused for multiple orderings.
 
+    def get_target(longest_path: bool = False) -> Key:
+        # Some topologies benefit if the node with the most dependencies
+        # is used as first choice, others benefit from the opposite.
+        candidates = leaf_nodes
+        skey: Callable = sort_key
+
+        preferred_candidates = set()
+        if runnable_hull:
+            skey = lambda k: (num_needed[k], sort_key(k))
+            preferred_candidates = runnable_hull & candidates
+        if not preferred_candidates and reachable_hull:
```

Comment:
with https://github.com/dask/dask/pull/11305 `runnable_hull & candidates` will always be empty, i.e. this clause can go away and the next if can be simplified

```suggestion
        if reachable_hull:
```

Reply:
Yep already did that on the rebase, updated now